pkg/argo_client/manifests.go

```diff
@@ -1,24 +1,84 @@
 package argo_client
 
 import (
+	"bytes"
 	"context"
+	"encoding/base64"
 	"fmt"
+	"os"
+	"os/exec"
+	"strconv"
+	"strings"
 	"time"
 
 	"github.com/argoproj/argo-cd/v2/pkg/apiclient/cluster"
 	"github.com/argoproj/argo-cd/v2/pkg/apiclient/settings"
 	argoappv1 "github.com/argoproj/argo-cd/v2/pkg/apis/application/v1alpha1"
 	repoapiclient "github.com/argoproj/argo-cd/v2/reposerver/apiclient"
 	"github.com/argoproj/argo-cd/v2/reposerver/repository"
+
 	"github.com/argoproj/argo-cd/v2/util/git"
 	"github.com/ghodss/yaml"
 	"github.com/pkg/errors"
 	"github.com/rs/zerolog/log"
 	"k8s.io/apimachinery/pkg/api/resource"
 
 	"github.com/zapier/kubechecks/telemetry"
+
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/service/ecr"
 )
 
+// Retrieve token for authentication against ECR registries.
+func getToken(aws_ecr_host string) (string, error) {
+	os.Setenv("AWS_SDK_LOAD_CONFIG", "1")
+	var region = strings.SplitN(string(aws_ecr_host), ".", 6)
+	cfg, err := config.LoadDefaultConfig(context.TODO(), config.WithRegion(region[3]))
+	if err != nil {
+		return "", err
+	}
+
+	svc := ecr.NewFromConfig(cfg)
+	token, err := svc.GetAuthorizationToken(context.TODO(), &ecr.GetAuthorizationTokenInput{})
+	if err != nil {
+		return "", err
+	}
+
+	authData := token.AuthorizationData[0].AuthorizationToken
+	data, err := base64.StdEncoding.DecodeString(*authData)
+	if err != nil {
+		return "", err
+	}
+
+	parts := strings.SplitN(string(data), ":", 2)
+
+	return parts[1], nil
+}
+
+func helmLogin(tempRepoDir string, changedAppFilePath string) error {
+
+	var aws_ecr_host = os.Getenv("AWS_ECR_HOST")
+	var currToken = ""
+	if token, err := getToken(aws_ecr_host); err != nil {
+		fmt.Println(err)
```

Comment:
we'll want to return this error, as we won't want to login to the registry if we can't get a token